src/components/views/dialogs/ResetIdentityDialog.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2025 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only OR GPL-3.0-only OR LicenseRef-Element-Commercial
+ * Please see LICENSE files in the repository root for full details.
+ */
+
+import React, { type JSX, type MouseEventHandler } from "react";
+
+import { MatrixClientPeg } from "../../../MatrixClientPeg";
+import MatrixClientContext from "../../../contexts/MatrixClientContext";
+import { ResetIdentityBody, type ResetIdentityBodyVariant } from "../settings/encryption/ResetIdentityBody";
+
+interface ResetIdentityDialogProps {
+    /**
+     * Called when the dialog closes.
+     */
+    onFinished?: () => void;
+
+    /**
+     * Called when the identity is reset (before onFinished is called).
+     */
+    onReset?: () => void;
+
+    /**
+     * Called when the reset is cancelled (before onFinished is called).
+     *
+     * Note: this is only called when the "Cancel" button is clicked, and not if
+     * the user clicks outside out the dialog, as that is handled by `Modal`
+     * rather than this component.
+     */
+    onCancelled?: () => void;
```

Comment:
sorry, but if we aren't going to use `onFinished` and `onCancelled`, which interact in a somewhat confusing way, I think we should probably get rid of them and the wrappers that make them work.